plugins/hls-gadt-plugin/src/Ide/Plugin/GHC.hs

```diff
@@ -0,0 +1,303 @@
+{-# LANGUAGE CPP              #-}
+{-# LANGUAGE GADTs            #-}
+{-# LANGUAGE LambdaCase       #-}
+{-# LANGUAGE PatternSynonyms  #-}
+{-# LANGUAGE RecordWildCards  #-}
+{-# LANGUAGE TupleSections    #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+module Ide.Plugin.GHC where
```

Comment:
I haven't really reviewed this, it seems complicated and we should probably get one of our exactprint experts to take a look.

Reply:
I'm not an exactprint expert either, but I have no concerns about the code in this module since it is well commented and isolated in this plugin